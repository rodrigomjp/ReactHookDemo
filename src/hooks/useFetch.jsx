import { useState, useEffect, useRef } from "react";

export const useFetch = (url) => {
  const isMounted = useRef(true);

  const [state, setState] = useState({
    data: null,
    loading: true,
    error: null,
  });
  //   useEffect(() => {
  //     return () => {
  //       //se ejecuta cuando se desmonta
  //       isMounted.current = false; //para indicar que se desmonto
  //     };
  //   }, []); //sÃ³lo se ejecute una vez al inicio

  useEffect(() => {
    //para limpiar el state antes de cada llamada
    setState({
      data: null,
      loading: true,
      error: null,
    });
    fetch(url)
      .then((resp) => resp.json())
      .then((data) => {
        if (isMounted.current) {
          setState({
            data: data,
            loading: false,
            error: null,
          });
        }
      });
  }, [url]);

  return state;
};
