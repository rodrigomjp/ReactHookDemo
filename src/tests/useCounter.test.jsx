import React from "react";
import { shallow } from "enzyme";
import { useCounter } from "../hooks/useCounter";
import { renderHook, act } from "@testing-library/react-hooks";

//para html
describe("Pruebas en <useCounter/>", () => {
  test("debe mostrase correctamente", () => {
    const wrapper = shallow(<useCounter></useCounter>);
    expect(wrapper).toMatchSnapshot();
  });
});

//para hook
describe("Pruebas en <useCounter/>", () => {
  test("debe mostrase correctamente", () => {
    const { result } = renderHook(() => useCounter());
    expect(result.current.counter).toBe(1);
    expect(typeof result.current.increment).toBe("function");
  });
});

describe("Pruebas en <useCounter/>", () => {
  test("debe incrmentar en 1 valor actual", () => {
    const { result } = renderHook(() => useCounter(1));
    const { increment } = result.current;
    act(() => {
      increment();
    });
    const { counter } = result.current;
    expect(counter).toBe(2);
  });
});

describe("Pruebas en <useCounter/>", () => {
  test("debe establecer el valor en 1", () => {
    const { result } = renderHook(() => useCounter(1));
    const { increment, reset } = result.current;
    act(() => {
      increment();
      reset();
    });
    const { counter } = result.current;
    expect(counter).toBe(1);
  });
});
