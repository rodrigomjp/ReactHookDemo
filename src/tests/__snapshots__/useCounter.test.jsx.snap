// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <useCounter/> debe mostrase correctamente 1`] = `<useCounter />`;
